bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -0,0 +1,278 @@
+ * sent to one of the channels.
+ *
+ * @author Christian Feininger - Initial contribution
+ */
+@NonNullByDefault
+public class SonnenHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SonnenHandler.class);
+
+    private SonnenConfiguration config = new SonnenConfiguration();
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private SonnenJSONCommunication serviceCommunication;
+
+    private boolean automaticRefreshing = false;
+
+    private Map<String, Boolean> linkedChannels = new HashMap<>();
+
+    public SonnenHandler(Thing thing) {
+        super(thing);
+        serviceCommunication = new SonnenJSONCommunication();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing sonnen handler for thing {}", getThing().getUID());
+        config = getConfigAs(SonnenConfiguration.class);
+        boolean validConfig = true;
+        String statusDescr = null;
+        if (config.refreshInterval < 0 && config.refreshInterval > 1000) {
+            statusDescr = "Parameter 'refresh Rate' greater then 0 and less then 1001.";
+            validConfig = false;
+        }
+        if (config.hostIP == null) {
+            statusDescr = "IP Address must be configured!";
+            validConfig = false;
+        }
+
+        Helper message = new Helper();
+        message.setStatusDescription(statusDescr);
+        if (validConfig) {
+            serviceCommunication.setConfig(config);
+            scheduler.submit(() -> {
+                if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+                    updateStatus(ThingStatus.ONLINE);
+                    updateLinkedChannels();
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            message.getStatusDesription());
+                }
+            });
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, message.getStatusDesription());
+        }
+    }
+
+    /**
+     * Calls the service to update the battery data
+     *
+     */
+    private boolean updatebatteryData() {
+        Helper message = new Helper();
+        if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+            updateStatus(ThingStatus.ONLINE);
+            return true;
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    message.getStatusDesription());
+            return false;
+        }
+    }
+
+    private void updateLinkedChannels() {
+
+        for (Channel channel : getThing().getChannels()) {
+            verifyLinkedChannel(channel.getUID().getId());
+        }
+
+        if (!linkedChannels.isEmpty()) {
+            updatebatteryData();
```

Comment:
Call to be removed as it will be called just after by startAutomaticRefresh()

Reply:
corrected